layers/+lang/lean/config.el

```diff
@@ -0,0 +1,16 @@
+;;; config.el --- Lean Layer config File for Spacemacs
+;;
+;; Copyright (c) 2012-2014 Sylvain Benner
+;; Copyright (c) 2014-2015 Sylvain Benner & Contributors
+;;
+;; Author: Sylvain Benner <sylvain.benner@gmail.com>
+;; URL: https://github.com/syl20bnr/spacemacs
+;;
+;; This file is not part of GNU Emacs.
+;;
+;;; License: GPLv3
+
+(configuration-layer/declare-layers '(auto-completion lua syntax-checking))
+
```

Comment:
Oh I see. Generally these are optional packages hence I mentioned. If lean-mode throws error or doesn't work then in that case declaring dependent layer might make sense. But I'll let others comment on that.